src/mail/editor/MailEditor.js

```diff
@@ -655,20 +662,34 @@ export async function newMailtoUrlMailEditor(mailtoUrl: string, confidential: bo
 
 	if (mailTo.attach) {
 		const attach = mailTo.attach
-		dataFiles = (await Promise.all((attach).map(uri => fileController.getDataFile(uri))))
+		dataFiles = (await Promise.all((attach).map(uri => fileController.getDataFile(uri)))).filter(Boolean)
 
 		// make sure the user is aware that (and which) files have been attached
-		const keepAttachments = dataFiles.filter(Boolean).length === 0 || await Dialog.confirm(
+		const keepAttachments = dataFiles.length === 0 || await Dialog.confirm(
 			"attachmentWarning_msg",
 			"attachFiles_action",
-			() => dataFiles.map((df, i) => df && m(".text-break.selectable.mt-xs", {
+			() => dataFiles.map((df, i) => m(".text-break.selectable.mt-xs", {
 				title: attach[i]
-			}, df.name)).filter(Boolean)
+			}, df.name))
 		)
 
-		dataFiles = keepAttachments
-			? dataFiles.filter(Boolean)
-			: []
+		const acceptedDataFiles = []
```

Comment:
we should put this in a function instead of copy pasting. Also, imo we should just not attach anything if the total size is too large, maybe we should do this in `SendMailModel` as well. I think it makes more sense for the user to choose which files to discard rather than arbitrarily discarding them ourselves

Reply:
I put the code in a function and use it in all three places (also in `ContactFormRequestDialog`). I am not sure I want to change the behavior of how files are attached or discarded in this bug fix. I suggest you open a new issue tagged as improvement in case you want to further discuss such a change.